storage/src/main/java/com/tencent/rss/storage/handler/impl/LocalFileServerReadHandler.java

```diff
@@ -206,45 +123,38 @@ private LocalFileReader createFileReader(String path) throws Exception {
     return new LocalFileReader(path);
   }
 
-  // index maybe change after first read, check index file size before read and update segments if necessary
-  private synchronized void updateIndexSegments() {
+  @Override
+  public ShuffleDataResult getShuffleData(long offset, int length) {
+    byte[] readBuffer = new byte[0];
+
     try {
-      File indexFile = new File(indexFileName);
-      long latestSize = indexFile.length();
-      if (latestSize > lastIndexFileSize) {
-        long start = System.currentTimeMillis();
-        initIndexReadSegment();
-        LOG.debug("Init index segments for " + indexFileName + " cost "
-            + (System.currentTimeMillis() - start) + " ms");
-        lastIndexFileSize = latestSize;
+      long start = System.currentTimeMillis();
+      try (LocalFileReader reader = createFileReader(dataFileName)) {
+        readBuffer = reader.readData(offset, length);
       }
+      LOG.debug(
+          "Read File segment: {}, offset[{}], length[{}], cost: {} ms, for appId[{}], shuffleId[{}], partitionId[{}]",
+          dataFileName, offset, length, System.currentTimeMillis() - start, appId, shuffleId, partitionId);
     } catch (Exception e) {
-      LOG.error("Error happened when update index segments", e);
+      LOG.warn("Can't read data for{}, offset[{}], length[{}]", dataFileName, offset, length);
```

Comment:
why not error？

Reply:
This may not be an error if this handler used in multi.